core/src/test/java/com/yahoo/gondola/rc/RcChannel.java

```diff
@@ -197,7 +198,9 @@ public void write(byte[] b, int off, int len) throws IOException {
                 }
             } catch (Exception e) {
                 message.release();
-                throw new IOException("key=" + key, e);
+                if (!"Read end dead".equals(e.getMessage())) {
```

Comment:
unfortunately there is no way to avoid this hardcoded string.
The only thing we can do is to create a test to make sure the string doesn't change.
Let me try that....